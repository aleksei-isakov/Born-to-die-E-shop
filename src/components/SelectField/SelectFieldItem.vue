<template>
  <li @mousedown="onClickChooseItem">
    {{ category }}
  </li>
</template>

<script>
export default {
  name: 'SelectFieldItem',

  props: {
    category: {
      type: String,
      required: true,
      default: ''
    },

    index: {
      type: Number,
      required: true,
      default: 0
    }
  },

  methods: {
    onClickChooseItem() {
      this.$emit('chooseItem', this.index);
    }
  }
};
</script>
