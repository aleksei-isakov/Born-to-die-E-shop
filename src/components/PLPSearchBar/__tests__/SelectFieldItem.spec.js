import SelectFieldItem from '../SelectFieldItem.vue';
import { mount } from '@vue/test-utils';

let wrapper;

beforeEach(() => {
  wrapper = mount(SelectFieldItem, {
    props: {
      category: 'Example of a category',
      index: 0
    }
  });
});

afterEach(() => {
  wrapper.destroy();
});
describe('SelectFieldItem', () => {
  it('should emit chooseItem event', () => {
    wrapper.vm.onClickChooseItem();
    expect(wrapper.emitted().chooseItem).toBeTruthy();
  });
  it('should render correctly and match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
