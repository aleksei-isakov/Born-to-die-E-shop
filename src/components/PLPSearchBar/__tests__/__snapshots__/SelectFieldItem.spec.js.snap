// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectFieldItem should render correctly and match snapshot 1`] = `
<li>
  
  

</li>
`;
